### ISSUE TITLE
Doesn't execute /etc/init.d/rcS on boot

### ISSUE THREAD

**Issue by Treebro** [maxnet/berryboot#104]
This is the intended default behaviour of buildroot, though you replace the init script with your own. I've  found your init script at: buildroot-2012.05/package/berrybootgui2/init

To workaround this so that buildroots added packages still funciton normally, have your init script execute /etc/init.d/rcS, which in turn executes all buildroot added packages or custom scripts starting with [S??] within the init.d directory.

Code exerpt:
  ifconfig
fi

/etc/init.d/rcS
# Show GUI, it will write the OS choosen to /tmp/answer

/usr/bin/BerrybootGUI -qws 2>/tmp/debug

Also sorry I'm not very good with git.

**Comment by maxnet** [comment]
Cannot make that the default, as the default Buildroot sysv init scripts will do things like activate the network and get a DHCP lease.
Folks that just want to have the boot menu might not want to wait for that.

**Comment by Treebro** [comment]
Are there any plans on adding an easily editable startup script? 

It would make sense to me to have it on the mmcblk0p1 partition so that it can be modified easily by plugging into any sd card reader.

**Comment by maxnet** [comment]
I doubt many folks start their own things under Berryboot.
What do you plan to start in the startup script?

**Comment by Treebro** [comment]
tl;dr: easy way to update, use it for integrity checks

My to do list on battling weird corruption issues for our in field pis:

For diagnosing issues
- Change root password 
- Create ssh access to root (internal network only, risk is irrelevant)
- explore telnet as a backup in case the ssh fails to operate through corruption.

For making devices more robust:
--We are noticing that even though the partition table is corrupting in most cases, berryboot still launches far enough to get to the vnc, but can't change the boot device in cmdline.txt because it will not mount.--
- Have script to check if paritition mounted correctly and if not, attempt to reformat to old sfdisk values, retaining data, (haven't looked into it much) or jump to berryboots initial install screen or format utility. ( This would not be able to be done in sd card mount, but rootfs itself. )
- On the fly updates that can be solved by a few startup commands.
- We've just implemented the 3.12.y update and we will see if this solves most of our issues, but will take  weeks -> months to see any useful results.

Current changes:
- Changed the size sfdisk formats
- Will not error out and exit if it cannot read its data partition, it will still go to mainwindow.  This has proven useful because I was able to edit cmdline.txt from the berryboot gui after it fails to boot, reformat the card, and be up and running again.
- Changed the way it searches for devices, includes mmcblk0p2 as a storage device to hold images on.  Very nice when booting to usb drive and store the images on sd card.  This has been made less useful now that I've changed the format size and the image can fit in mmcblk0p1. Files can be loaded from /boot in the berryboot gui.

**Comment by maxnet** [comment]
While there is currently no way to add startup scripts, you do can make shell scripts a menu item in Berryboot, and create menu choices for "Enable SSH server" "Reformat disk", etc. that way.

For that create a squashfs image containing a single file named "berryboot-init"
When started, the contents of berryboot-init will be executed as shell commands under the Berryboot environement.

**Comment by Treebro** [comment]
As far as this ticket goes this isn't an issue anymore.

You'd have to walk me through adding menu items.  I don't think I quite understand much about Qt.
I'll keep working on my current path as I am confident I'll get it going.  I absolutely need the ssh to start autonomously and the reformat would be fine as a menu item.   Is there no way to have conversations over this site as private to avoid cluttering / issue tickets?

**Comment by maxnet** [comment]
> As far as this ticket goes this isn't an issue anymore.

Ok, closing it then.

> You'd have to walk me through adding menu items. I don't think I quite understand much about Qt.

It isn't something Qt specific.
Just mentioning that in addition to having full blown "OS images" like Raspbian and OpenELEC in Berryboot, Berryboot also supports having "images" that only consists of a single shell script.

> I'll keep working on my current path as I am confident I'll get it going. 
> I absolutely need the ssh to start autonomously and the reformat would be fine as a menu item. Is there no way to have conversations over this site as private to avoid cluttering / issue tickets?

Don't think Github does private conversations.
My e-mail address is in the README though if you need it.
